alns/accept/tests/test_late_acceptance_hill_climbing.py

```diff
@@ -0,0 +1,157 @@
+import numpy.random as rnd
+from numpy.testing import assert_, assert_equal, assert_raises
+from pytest import mark
+
+from alns.accept import LateAcceptanceHillClimbing
+from alns.tests.states import Zero, One, Two
+
+
+@mark.parametrize("lookback_period", [-0.01, -10, 1.5])
+def test_raises_invalid_lookback_period(lookback_period):
+    with assert_raises(ValueError):
+        LateAcceptanceHillClimbing(lookback_period=lookback_period)
+
+
+@mark.parametrize(
+    "lookback_period, greedy, better_history",
+    [
+        (0, True, True),
+        (1, False, True),
+        (10, True, False),
+        (100, False, False),
+    ],
+)
+def test_properties(lookback_period, greedy, better_history):
+    lahc = LateAcceptanceHillClimbing(lookback_period, greedy, better_history)
+
+    assert_equal(lahc.lookback_period, lookback_period)
+    assert_equal(lahc.better_history, better_history)
+    assert_equal(lahc.greedy, greedy)
+
+
+def test_regular_hill_climbing():
+    """
+    Test if LAHC behaves like regular hill climbing when `lookback_period` is
+    set to zero.
+    """
+    lahc = LateAcceptanceHillClimbing(0, False, False)
+
+    assert_(lahc(rnd.RandomState(), Zero(), Two(), One()))
+    assert_(not lahc(rnd.RandomState(), Zero(), One(), One()))
+    assert_(not lahc(rnd.RandomState(), Zero(), Zero(), Zero()))
+    assert_(lahc(rnd.RandomState(), Zero(), Two(), One()))
+
+
+@mark.parametrize("lookback_period", [3, 10, 50])
```

Comment:
Test: can we add the case `lookback_period = 0` to this test, and other relevant parametrised tests? There are several in this file.